index.js

```diff
@@ -370,20 +370,67 @@ function isWebcam(device) {
     device.deviceDescriptor.bDeviceProtocol === 0x01
 }
 
+function getInterfaces(device) {
+  const interfaces = Array.isArray(device.interfaces) ? device.interfaces : [];
+  device.allConfigDescriptors.forEach(config => {
```

Comment:
Based on the example below, allConfigDescriptors is not present on Mac, so you should use optional chaining.
```
device.allConfigDescriptors?.forEach(iface => {
```